.github/actions/install-go-and-dependencies/action.yml

```diff
@@ -0,0 +1,18 @@
+name: InstallGo and configure aws creds
```

Comment:
We are not reusing this. Why don't we embed it?

Reply:
Test coverage for kit guest cluster and Eks guest cluster also needs installation of GO,Instead of re writing the installation code again we can reuse it.